zkml/docs/src/end_to_end_lu.md

```diff
@@ -0,0 +1,28 @@
+# End-to-End Lookup Protocol
+Since a model will have many repeated layers to improve efficiency we aim to only produce one multiplicity polynomial per table type. An added bonus to this is that at each step we only have to do proving work proportional to the number of lookups (for lookup steps) or the table size (for mulitpilcity polynomials).
```

Comment:
I dont see the "trick" in this doc. 
Also can you clarify, with the trick explanation, what polynomials do we commit to, when and how ?